engines/director/archive.cpp

```diff
@@ -640,21 +641,35 @@ bool RIFXArchive::openStream(Common::SeekableReadStream *stream, uint32 startOff
 	delete dumpStream;
 
 	// If we couldn't read the map, we can't do anything past this point.
-	if (!readMapSuccess)
+	if (!readMapSuccess) {
+		_stream = nullptr;
 		return false;
+	}
 
 	if (_rifxType == MKTAG('A', 'P', 'P', 'L')) {
 		if (hasResource(MKTAG('F', 'i', 'l', 'e'), -1)) {
 			// Replace this archive with the embedded archive.
-			uint32 fileId = getResourceIDList(MKTAG('F', 'i', 'l', 'e'))[0];
-			int32 fileOffset = _resources[fileId]->offset;
-			_types.clear();
-			_resources.clear();
-			return openStream(_stream, fileOffset);
-		} else {
-			warning("No 'File' resource present in APPL archive");
-			return false;
+			Common::Array<uint16> subFiles = getResourceIDList(MKTAG('F', 'i', 'l', 'e'));
+			uint numFile;
+			int32 fileOffset = 0;
+			for (numFile = 0; numFile < subFiles.size(); numFile++) {
+				uint32 fileId = subFiles[numFile];
+				fileOffset = _resources[fileId]->offset;
+				endianStream.seek(fileOffset + 8);
+				uint32 tag = endianStream.readUint32();
+				if (tag != MKTAG('X', 't', 'r', 'a') && tag != MKTAG('a', 'r', 't', 'X'))
```

Comment:
Since the stream is endian, why checking for different endianness?

Reply:
Mostly for safety.  They are containers inside of containers so I didn't want to assume that both levelsuse the same endianness

Reply:
Looks good, thank you!